pkg/collector/corechecks/snmp/internal/profile/profile_initconfig.go

```diff
@@ -0,0 +1,41 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package profile
+
+import "github.com/DataDog/datadog-agent/pkg/util/log"
+
+func loadInitConfigProfiles(rawInitConfigProfiles ProfileConfigMap) (ProfileConfigMap, error) {
+	initConfigProfiles := make(ProfileConfigMap, len(rawInitConfigProfiles))
+
+	for name, profConfig := range rawInitConfigProfiles {
+		if profConfig.DefinitionFile != "" {
+			profDefinition, err := readProfileDefinition(profConfig.DefinitionFile)
+			if err != nil {
+				log.Warnf("failed to read profile definition `%s`: %s", name, err)
+				continue
+			}
+			profConfig.Definition = *profDefinition
+		}
+		initConfigProfiles[name] = profConfig
+	}
+
+	userProfiles := mergeProfiles(initConfigProfiles, getYamlUserProfiles())
```

Comment:
Indeed, but we still need to user the user+default yaml profiles for resolving extends when applicable.

This is the previous behaviour that we are keeping (to avoid breaking change).